lib/content.rb

```diff
@@ -6,19 +6,43 @@ def self.slugs_by_page_uuid
       "slugs_by_page_uuid/#{OpenStax::Content::Abl.new.digest}"
     ) do
       {}.tap do |hash|
-        OpenStax::Content::Abl.new.approved_books.each do |book|
-          begin
-            pages = book.all_pages
-          rescue StandardError => exception
-            # Sometimes books in the ABL fail to load
-            # Log it, but keep going
-            Rails.logger.error exception
-          else
-            pages.each do |page|
-              hash[page.uuid] ||= []
-              hash[page.uuid] << { book: book.slug, page: page.slug }
+        books = OpenStax::Content::Abl.new.approved_books
+
+        until books.empty?
+          previous_version = nil
+          previous_archive = nil
+          retry_books = []
+
+          books.each do |book|
+            begin
+              pages = book.all_pages
+            rescue StandardError => exception
+              # Sometimes books in the ABL fail to load
+              # Retry with an earlier version of archive, if possible
+              previous_version ||= book.archive.previous_version
+
+              if previous_version.nil?
+                raise exception
```

Comment:
I like re-raising here instead of logging, though I wonder if there will be a case where the ABL randomly fails to load and there really is only 1 version. Maybe that's not a likely scenario though.

Reply:
I think that would break Rex pretty badly, but it's a possibility if someone is too eager to update the ABL. Probably better that it notifies us though.